src/padding.rs

```diff
@@ -1,12 +1,166 @@
-// Applies new padding to a BSON packet.
-// Returns the padded packet.
-fn encapsulate(pkt: &[u8]) -> Vec<u8> {
-    unimplemented!()
+use rand::{distributions::Uniform, CryptoRng, Rng, SeedableRng};
+use std::io::{Error, ErrorKind};
+
+/// Generates a random number between 2 and 32 (inclusive) with the given random number generator.
+///
+/// # Arguments
+///
+/// * `rng` - A mutable reference to a random number generator.
+///
+/// # Returns
+///
+/// A random u32 between 2 and 32 (inclusive).
+
+fn generate_random_num<R: Rng + CryptoRng>(rng: &mut R) -> u32 {
+    let range = Uniform::new_inclusive(2, 32);
+    rng.sample(range)
+}
+
+/// Generates a vector of random bytes with the specified length, excluding a specific character.
+///
+/// # Arguments
+///
+/// * `rng` - A mutable reference to a random number generator.
+/// * `count` - The number of bytes to generate.
+/// * `exclusion` - The character to exclude from the generated bytes.
+///
+/// # Returns
+///
+/// A vector of random bytes with the defined length, excluding the specified character.
+
+fn generate_random_bytes<R: Rng>(rng: &mut R, count: u32, exclusion: char) -> Vec<u8> {
+    let mut bytes = Vec::with_capacity(count as usize);
+
+    // Use the random number generator to fill the defined buffer with random bytes.
+    let mut buf = [0u8; 256];
+    rng.try_fill_bytes(&mut buf)
+        .expect("Failed to fill buffer.");
+
+    // Iterate over the buffer to generate random bytes.
+    for &byte in &buf {
+        // Check if the desired number of bytes has been generated.
+        if bytes.len() == count as usize {
+            break;
+        }
+
+        // Exclude the specified character from the generated bytes
+        if byte != exclusion as u8 {
+            bytes.push(byte);
+        }
+    }
+
+    // Return the vector of random bytes.
+    bytes
+}
+
+/// Encapsulates the given BSON packet with a random number (between 2 and 32 inclusive) of random bytes before and after.
+///
+/// # Arguments
+///
+/// * `pkt` - A reference to a BSON packet.
+///
+/// # Returns
+///
+/// The encapsulated packet in Vec<u8> form.
+///
+/// # Errors
+///
+/// Returns an error if the BSON packet is not valid (it does not start and end with the correct braces) or
+/// if there is an error during the generation of random bytes or packet encapsulation
+
+fn encapsulate(pkt: &[u8]) -> Result<Vec<u8>, std::io::Error> {
+    // if the packet does not start and end with braces, it is not valid BSON data
+    if pkt[0] != b'{' || pkt[pkt.len() - 1] != b'}' {
+        return Err(Error::from(ErrorKind::InvalidData));
+    }
+
+    let mut rng = rand_chacha::ChaCha8Rng::from_entropy(); // use the ChaCha8Rng algorithm to generate our random numbers and bytes
```

Comment:
This was entirely unnecessary - we already have `rand` which should pull from entropy, we don't need ChaCha's randomness, especially since this is for our padding and not for cryptographic purposes.